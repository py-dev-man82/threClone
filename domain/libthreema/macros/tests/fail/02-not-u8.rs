use libthreema_macros::concat_fixed_bytes;

fn main() {
    let a = [1u8; 4];
    let b = [2u8; 3];
    let c = [3u8; 3];
    let d = [4u16; 2];

    let result: [u8; 4] = concat_fixed_bytes!(d); // d is a u16 array
    let result: [u8; 8] = concat_fixed_bytes!(a, d); // d is a u16 array
}
