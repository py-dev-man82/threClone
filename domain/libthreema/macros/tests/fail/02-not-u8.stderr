error[E0308]: mismatched types
 --> tests/fail/02-not-u8.rs:9:47
  |
9 |     let result: [u8; 4] = concat_fixed_bytes!(d); // d is a u16 array
  |                                               ^ expected `[u8; _]`, found `[u16; 2]`
  |
  = note: expected array `[u8; _]`
             found array `[u16; 2]`

error[E0308]: mismatched types
  --> tests/fail/02-not-u8.rs:10:50
   |
10 |     let result: [u8; 8] = concat_fixed_bytes!(a, d); // d is a u16 array
   |                                                  ^ expected `[u8; _]`, found `[u16; 2]`
   |
   = note: expected array `[u8; _]`
              found array `[u16; 2]`
